drivers/sensor/ti/tmp435/tmp435.c

```diff
@@ -0,0 +1,216 @@
+			return ret;
+		}
+		temp = value;
+		ret = tmp435_reg_read(cfg, TMP435_LOCAL_TEMP_L_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_LOCAL_TEMP_L_REG, ret:%d", ret);
+			return ret;
+		}
+		if (value > TMP435_FRACTION_INC) {
+			temp++;
+		}
+		data->temp_die = temp + tmp435_temp_offset;
+	}
+
+	if ((chan == SENSOR_CHAN_ALL) || (chan == SENSOR_CHAN_AMBIENT_TEMP)) {
+		if (!(cfg->external_channel)) {
+			return 0; /* not enabled, just return */
+		}
+		ret = tmp435_reg_read(cfg, TMP435_REMOTE_TEMP_H_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_REMOTE_TEMP_H_REG ret:%d", ret);
+			return ret;
+		}
+		temp = value;
+		ret = tmp435_reg_read(cfg, TMP435_REMOTE_TEMP_L_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_REMOTE_TEMP_L_REG, ret:%d", ret);
+			return ret;
+		}
+		if (value > TMP435_FRACTION_INC) {
+			temp++;
+		}
+		data->temp_ambient = temp + tmp435_temp_offset;
+	}
+	return 0;
+}
+
+static int tmp435_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	int ret = 0;
+	struct tmp435_data *data = dev->data;
+	const struct tmp435_config *cfg = dev->config;
+
+	switch (chan) {
+
+	case SENSOR_CHAN_DIE_TEMP:
+		val->val1 = data->temp_die;
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		if (cfg->external_channel) {
+			val->val1 = data->temp_ambient;
+			val->val2 = 0;
+		} else {
+			ret = -ENOTSUP;
+		}
+		break;
+	default:
+		ret = -ENOTSUP;
+		break;
+	}
+	return ret;
+}
+
+static const struct sensor_driver_api tmp435_driver_api = {
+	.sample_fetch = tmp435_sample_fetch,
+	.channel_get = tmp435_channel_get,
+};
+
+static int tmp435_init(const struct device *dev)
+{
+	uint8_t data = 0;
+	int ret = 0;
+	const struct tmp435_config *cfg = dev->config;
+
+	if (!(i2c_is_ready_dt(&cfg->i2c))) {
+		LOG_ERR("I2C dev not ready");
+		return -ENODEV;
+	}
+
+	data = 1; /* write anything to reset */
+	ret = tmp435_reg_write(cfg, TMP435_SOFTWARE_RESET_REG, &data, 1);
+	if (ret < 0) {
+		LOG_ERR("Failed to write TMP435_SOFTWARE_RESET_REG ret:%d", ret);
+		return ret;
+	}
+
+	data = TMP435_CONF_REG_1_DATA;
+	ret = tmp435_reg_write(cfg, TMP435_CONF_REG_1, &data, 1);
+	if (ret < 0) {
+		LOG_ERR("Failed to write TMP435_CONF_REG_1 ret:%d", ret);
+		return ret;
+	}
+
+	data = TMP435_CONF_REG_2_DATA;
+	if (cfg->external_channel) {
+		data = data + TMP435_CONF_REG_2_REN;
+	}
+	if (cfg->resistance_correction) {
+		data = data + TMP435_CONF_REG_2_RC;
+	}
+	ret = tmp435_reg_write(cfg, TMP435_CONF_REG_2, &data, 1);
+	if (ret < 0) {
+		LOG_ERR("Failed to write TMP435_CONF_REG_2 ret:%d", ret);
+		return ret;
+	}
+
+	data = cfg->beta_compensation;
+	ret = tmp435_reg_write(cfg, TMP435_BETA_RANGE_REG, &data, 1);
+	if (ret < 0) {
+		LOG_ERR("Failed to write TMP435_BETA_RANGE_REG ret:%d", ret);
+		return ret;
+	}
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
```

Comment:
This seems redundant since it should not be possible to enable due to `depends on` in the Kconfig

Reply:
Requested changes done.

Reply:
Done